pkg/base/helm.go

```diff
@@ -74,6 +74,32 @@ func RenderHelm(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (*Base,
 		return nil, errors.Errorf("unknown helmVersion %s", renderOptions.HelmVersion)
 	}
 
+	// add namespace if it's defined in the spec and not already present in the manifests
+	if renderOptions.Namespace != "" {
+		for i, v := range rendered {
+			var manifest map[string]interface{}
+			err := yaml.Unmarshal(v.Content, &manifest)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to unmarshal helm manifest")
```

Comment:
if marshalling or unmarshalling fails, is it better to return an error or just skip adding the namespace?

Reply:
leaving the errors as is but let me know if something else is preferred